src/gwt/src/org/rstudio/core/client/command/ShortcutManager.java

```diff
@@ -443,7 +444,8 @@ public void execute()
       }
 
       // Bail if this is an ignored key combination.
-      if (isIgnoredKeyCombination(keyCombination))
+      if (isIgnoredKeyCombination(keyCombination) || 
+            XTermWidget.isXTerm(Element.as(event.getEventTarget())))
```

Comment:
This may be over-aggressive -- I think this will imply that commands which have a more global scope (e.g. `Build and Reload`) won't work while an XTermWidget has focus.

That said, I'm not sure what the best way to white-list (black-list?) commands here would be -- maybe some extra metadata in `commands.xml`?